Src/Common/CrossComponentCorrelationEventSource.cs

```diff
@@ -55,13 +57,27 @@ public void GetHeaderFailed(string exception, string appDomainName = "Incorrect"
             this.WriteEvent(3, exception, this.ApplicationName);
         }
 
+        [Event(
+            4,
+            Keywords = Keywords.Diagnostics,
+            Message ="Unknown error occurred.",
+            Level = EventLevel.Warning)]
+        public void UnknownError(string exception, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(4, exception, this.ApplicationName);
+        }
+
         [NonEvent]
         private string GetApplicationName()
         {
             string name;
             try
             {
+#if NETCORE
+                name = typeof(CrossComponentCorrelationEventSource).GetTypeInfo().Assembly.GetName().FullName;
```

Comment:
It's supposed to be the equivalent of AppDomain.CurrentDomain.FriendlyName. .NETStandard doesn't support AppDomain, so this seemed to be the closest I could find, but it sounds like it's still not right. Do you have any suggestions?